File: Zig Zag Sequence/ZigZag.py
def findZigZagSequence(a, n):
    a.sort()  # Sort the array
    mid = (n - 1) // 2  # Correct middle index
    a[mid], a[n - 1] = a[n - 1], a[mid]  # Swap middle with the last element

    # Reverse only the elements from mid+1 to n-2
    st = mid + 1
    ed = n - 2
    while st < ed:  # Use strict < to avoid unnecessary swaps
        a[st], a[ed] = a[ed], a[st]
        st += 1
        ed -= 1

    # Print the zig-zag sequence
    for i in range(n):
        if i == n - 1:
            print(a[i])
        else:
            print(a[i], end=' ')
    return

# Main logic to handle input/output
t = int(input())
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    findZigZagSequence(a, n)


# the code above works but the problem only wants you to edit 3 lines total

# Here is the original
'''
def findZigZagSequence(a, n):
    a.sort()
    mid = int((n + 1)/2)
    a[mid], a[n-1] = a[n-1], a[mid]

    st = mid + 1
    ed = n - 1
    while(st <= ed):
        a[st], a[ed] = a[ed], a[st]
        st = st + 1
        ed = ed + 1

    for i in range (n):
        if i == n-1:
            print(a[i])
        else:
            print(a[i], end = ' ')
    return

test_cases = int(input())
for cs in range (test_cases):
    n = int(input())
    a = list(map(int, input().split()))
    findZigZagSequence(a, n)
'''

# Here is the Corrected version
'''
def findZigZagSequence(a, n):
    a.sort()
    mid = (n - 1) // 2 # Corrected middle index calculation
    a[mid], a[n-1] = a[n-1], a[mid]

    st = mid + 1
    ed = n - 2 # Corrected the end index calculation
    while(st <= ed):
        a[st], a[ed] = a[ed], a[st]
        st = st + 1
        ed = ed - 1 # Corrected the decrement operation

    for i in range (n):
        if i == n-1:
            print(a[i])
        else:
            print(a[i], end = ' ')
    return

test_cases = int(input())
for cs in range (test_cases):
    n = int(input())
    a = list(map(int, input().split()))
    findZigZagSequence(a, n)
'''

File: Zig Zag Sequence/ZigZag.cs
using System;
using System.Linq;

class Solution
{
    static void Main(string[] args)
    {
        // Number of test cases
        int t = int.Parse(Console.ReadLine());
        for (int cs = 0; cs < t; cs++)
        {
            // Size of the array
            int n = int.Parse(Console.ReadLine());
            int[] a = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            FindZigZagSequence(a, n);
        }
    }

    static void FindZigZagSequence(int[] a, int n)
    {
        // Sort the array
        Array.Sort(a);

        // Find the middle index
        int mid = (n - 1) / 2;

        // Swap the middle element with the last element
        int temp = a[mid];
        a[mid] = a[n - 1];
        a[n - 1] = temp;

        // Reverse only the elements from mid+1 to the end of the array
        int st = mid + 1;
        int ed = n - 1;
        while (st < ed)
        {
            temp = a[st];
            a[st] = a[ed];
            a[ed] = temp;
            st++;
            ed--;
        }

        // Print the zig-zag sequence
        Console.WriteLine(string.Join(" ", a));
    }
}
